### DISCUSSION TITLE
Home timeline cuts out abruptly

### DISCUSSION THREAD

**Discussion by MirceaKitsune** [mastodon/mastodon#27909] [General]
I noticed on several instances that if you scroll the home timeline too much to the bottom, it stops loading in new posts. I don't know if this is a bug or a deliberate limitation for performance reasons, but it's been an issue several times in the past. If you're watching even a few active users, you often can't scroll to even one day old posts! It would be nice if this limitation could at least be raised, so the average user can keep scrolling down to a week's worth of posts.

An alternative idea, in case limiting the length of the home timeline is required for performance: Don't limit based on post count (unless it's extremely high) but based on post age. This would optimize the server by ensuring that people who watch fewer accounts have a smaller home timeline, while only those who watch a lot of people require processing a bigger one.

**Comment by ke352802081770314** [comment]
Interestingly, in "Discussions" this was the only one I found on this topic. There is a limit which was increased from 400 to 800 posts in v4.1.0 (see #2301 and #18908 in Issues).
I think such a hard limit is a major inconvenience, not on a daily basis but very severe when it catches you. I tried working around it by unfollowing various "chatty" accounts, but that did not make my /home timeline go back further! I also considered moving some accounts into lists with "hide from home timeline" but that would only work into the future so is not usable as a workaround when you want to recover past posts in your timeline Mastodon decides to no longer give you when you really need them.
